R/utilities-units.R

```diff
@@ -95,38 +95,22 @@ getBaseUnit <- function(dimension) {
 #' @export
 toBaseUnit <- function(quantityOrDimension, values, unit, molWeight = NULL, molWeightUnit = NULL) {
   validateIsOfType(quantityOrDimension, c("Quantity", "character"))
-  validateIsNumeric(values, nullAllowed = TRUE)
-  validateIsNumeric(molWeight, nullAllowed = TRUE)
-  unit <- .encodeUnit(unit)
-  dimension <- quantityOrDimension
-  dimensionTask <- .getNetTask("DimensionTask")
-
-  # covers all NULL or NA
-  if (all(is.na(values))) {
-    return(values)
-  }
-
-  # ensure that we are dealing with an list of values seen as number (and not integer)
-  values <- as.numeric(c(values))
 
+  # Get the base unit of the dimension and call `toUnit()`
+  dimension <- quantityOrDimension
   if (isOfType(quantityOrDimension, "Quantity")) {
     dimension <- quantityOrDimension$dimension
   }
-
-  if (all(is.na(molWeight))) {
-    molWeight <- NULL
-  }
-
-
-  if (is.null(molWeight)) {
-    return(rClr::clrCall(dimensionTask, "ConvertToBaseUnit", dimension, unit, values))
-  }
-
-  # Convert molWeight value to base unit if a unit is provided
-  if (!is.null(molWeightUnit)) {
-    molWeight <- rClr::clrCall(dimensionTask, "ConvertToBaseUnit", ospDimensions$`Molecular weight`, molWeightUnit, molWeight)
-  }
-  rClr::clrCall(dimensionTask, "ConvertToBaseUnit", dimension, unit, values, molWeight)
+  baseUnit <- getBaseUnit(dimension)
+
+  toUnit(
```

Comment:
Just get the base unit of the dimension and call `toUnit()`

Reply:
Nice